src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -125,6 +126,12 @@ internal class SourcePropertyAccessorSymbol : SourceMemberMethodSymbol
                 syntax,
                 diagnostics);
         }
+
+        private static bool NodeContainsFieldKeyword(CSharpSyntaxNode node)
+            => node.DescendantTokens().Any(
+                t => t.IsKind(SyntaxKind.IdentifierToken) && t.ContextualKind() == SyntaxKind.FieldKeyword && t.Parent.IsKind(SyntaxKind.IdentifierName));
```

Comment:
>t.Parent.IsKind(SyntaxKind.IdentifierName) [](http://example.com/codeflow?start=110&length=42)

We probably also want to check that this is not the right side of a QualifiedName, an AliasQualifiedName or a MemberAccess. Might be worth checking other syntax nodes that could use IdentifierName as a child.

Reply:
@AlekseyTs member access is tested in `TestAccessingNonExistingFieldMember` and seems to be working as expected. However, `NodeContainsFieldKeyword` is actually true for this case. Given that there is a test (and will add more tests for this scenario), do I still need to change the implementation?

Reply:
> Given that there is a test (and will add more tests for this scenario), do I still need to change the implementation?

I think yes. We don't have to change this in this PR (a PROTOTYPE comment will be good for now), but I think we want this syntactic check to be as precise as possible to avoid extra unnecessary work. A ```field``` identifier following a ```.``` or a ```->``` cannot be a keyword.